llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -29986,7 +29986,7 @@ static SDValue LowerShift(SDValue Op, const X86Subtarget &Subtarget,
           MVT::getVectorVT(NarrowScalarVT, WideNumElts), dl, AmtWideElts);
       AmtWide = DAG.getZExtOrTrunc(AmtWide, dl, WideVT);
       // Perform the actual shift.
-      unsigned LogicalOpc = Opc == ISD::SRA ? ISD::SRL : Opc;
+      unsigned LogicalOpc = Opc == ISD::SRA ? (unsigned)ISD::SRL : Opc;
```

Comment:
Why need `(unsigned)` here? If the `ISD::XXX` is not a unsigned value, I think we would have more warn/error when comparing with `Opc`.

Reply:
Oh, maybe because we just use `==` and `!=` here.

Reply:
It is about `Opc` is unsigned and `ISD::SRL` is enum, and they are used in the ?: operator, so type mismatch.